src/util/execute-lifecycle-script.js

```diff
@@ -116,8 +121,22 @@ export async function executeLifecycleScript(
   // split up the path
   const pathParts = (env[constants.ENV_PATH_KEY] || '').split(path.delimiter);
 
-  // add node-gyp
-  pathParts.unshift(path.join(__dirname, '..', '..', 'bin', 'node-gyp-bin'));
+  // Include node-gyp version that was bundled with the current Node.js version,
+  // if available.
+  pathParts.unshift(
+    path.join(
+      path.dirname(process.execPath), 'node_modules', 'npm', 'bin', 'node-gyp-bin',
+    ),
+  );
+  pathParts.unshift(
+    path.join(
```

Comment:
`path.join` gets rid of the `..` too 😃 

I used `path.join` with all the pieces rather than hard-coding a string with `/` since it seemed cleaner. I guess either way is fine.